src/main/java/uk/gov/pay/adminusers/service/StripeAgreementService.java

```diff
@@ -28,8 +29,8 @@ public StripeAgreementService(StripeAgreementDao stripeAgreementDao) {
                 .map((StripeAgreementEntity::toStripeAgreement));
     }
     
-    public void doCreate(int serviceId, String ipAddress, LocalDateTime agreementTime) {
+    public void doCreate(int serviceId, StripeAgreementRequest stripeAgreementRequest, LocalDateTime agreementTime) {
```

Comment:
I think I’d prefer if we didn’t pass the `StripeAgreementRequest` in here and instead passed in just the IP address (which could be stored as a `InetAddress` or something rather than a `String`). Ideally, a service shouldn’t really know that it’s part of a web application.